R/read_alevin.R

```diff
@@ -115,29 +106,101 @@ read_usa_mode <- function(quant_dir,
     stop(paste0("Missing Alevin output file(s): ", missing_files))
   }
 
+  # read in .mtx files
+  counts <- Matrix::readMM(file = file.path(quant_dir, "alevin", "quants_mat.mtx"))%>%
+    Matrix::t() %>%
+    as("dgCMatrix")
+  cols <- readLines(file.path(quant_dir, "alevin", "quants_mat_cols.txt"))
+  rows <- readLines(file.path(quant_dir, "alevin", "quants_mat_rows.txt"))
+  dimnames(counts) <- list(cols, rows)
+  return(counts)
+}
+
+#' Read in counts data processed with Alevin or alevin-fry in tximport-compatible formats
+#'
+#' @param quant_dir Path to alevin output directory.
+#'
+#' @return unfiltered & uncollapsed gene x cell counts matrix
+#'
+read_tximport <- function(quant_dir){
+
+  # check that all files exist in quant_dir
+  # use tximport for all non-usa mode
+  alevin_files <- c("quants_mat_cols.txt", "quants_mat_rows.txt", "quants_mat.gz")
+
+  missing <- !file.exists(file.path(quant_dir, "alevin", alevin_files))
+  if(any(missing)) {
+    missing_files <- paste(alevin_files[missing], collapse = ", ")
+    stop(paste0("Missing Alevin output file(s): ", missing_files))
+  }
+
+  txi <- suppressMessages(tximport::tximport(
+    file.path(quant_dir, "alevin", "quants_mat.gz"),
+    type = "alevin"
+  ))
+  counts <- txi$counts
+}
+
+#' Read alevin metadata from json files
+#'
+#' @param quant_dir Path alevin output directory.
+#'
+#' @return A list containing alevin run metadata,
+#'   with NULL values for missing elements.
+#'
+#' @noRd
+read_alevin_metadata <- function(quant_dir){
```

Comment:
```suggestion
read_alevin_metadata <- function(quant_dir, which_counts = c("spliced", "unspliced"){
```
If we want to add transcript type during this part why don't we include it as an argument here and then append it later?

Reply:
I thought about doing this (it was there in a pervious version) but I think my logic was that this function is (by its name) about reading metadata, so anything not read in could be added separately.